ADAL/tests/ADTestURLSessionDataTask.m

```diff
@@ -69,6 +69,11 @@ - (void)resume
         return;
     }
     
+    if (response->_waitSemaphore)
+    {
+        dispatch_semaphore_wait(response->_waitSemaphore, DISPATCH_TIME_FOREVER);
+    }
+    
```

Comment:
Ok, thanks for the information. Just one suggestion: it would be nice to add a comment to setWaitSemaphore: method that the test is then responsible for signaling the semaphore itself.